pydbus/strong_typing.py

```diff
@@ -34,3 +35,10 @@ def decorate(func):
 		verify_arguments(func)
 		return func
 	return decorate
+
+
+class TypedSignal(signal):
+
+	def __init__(self, arguments):
```

Comment:
Well for that there is the `@signal` decorator. For example:

```python
# In Python 3
@signal
def PropertiesChanged(interface_name: 's', changed_properties: 'a{sv}', invalidated_properties: 'as'):
    pass
```

```python
# In Python 2
@signal
@typed_method(("s", "a{sv}", "as"), None)
def PropertiesChanged(interface_name, changed_properties, invalidated_properties):
    pass
```

Now in both cases the decorator will store the method inside the `signal` class itself. And if the introspector comes along and sees a `signal` class it can access that method object and then do the almost same stuff it'd do to a normal method. The only difference is that it won't allow a return type (as far as I understand it that isn't a thing in D-Bus) and that no argument has a direction defined (or if it were it should be `out` not `in`).

As a matter of fact I have actually already implemented it that way and just finished with cleaning it up. So if you wanted I could upload it soonish. (As I write this I noticed that I haven't verified that `@interface` behaves as expected)

The only ugly thing there is the superfluous `pass` block. But I think apart from that it is actually quite nice.